File: 6.객체와_자료구조/test/6-1-test.py
import math


class Squre(object):
    x: float
    y: float
    side: float


class Rectangle(object):
    x: float
    y: float
    width: float
    height: float


class Circle(object):
    x: float
    y: float
    radius: float


def get_arae(figure: object) -> float:
    if isinstance(figure, Squre):
        return figure.side * figure.side

    elif isinstance(figure, Rectangle):
        return figure.width * figure.height

    elif isinstance(figure, Circle):
        return math.pi * figure.radius * figure.radius

    else:
        raise Exception('no such figure')

File: CleanCode/6.객체와_자료구조/test/6-1-test.go
package test

import (
	"math"
)

type Point struct {
	x float32
	y float32
}

type Figure interface {
	GetArea() float32
}

type Sqaure struct {
	point *Point
	side  float32
}

type Rectangle struct {
	point  *Point
	width  float32
	height float32
}

type Circle struct {
	point  *Point
	radius float32
}

func (s *Sqaure) GetArea() float32 {
	return s.side * s.side
}

func (r *Rectangle) GetArea() float32 {
	return r.width * r.height
}

func (c *Circle) GetArea() float32 {
	return math.Pi * c.radius * c.radius
}
